### PR TITLE
make a parallel shipcat validate (verify)

### PR DESCRIPTION
**Pull request by clux**
this will eventually deprecate validate.
can run verify on all regions in about 3s locally.

### CODE REVIEW

File: shipcat_cli/src/validate.rs [hunk 1]
```diff
@@ -1,5 +1,63 @@
...
+use std::sync::mpsc::channel;
+
+/// Validate all manifests in a service directory for a region
+///
+/// This is meant to replace `shipcat validate ..all_services`
+pub fn regional_manifests(conf: &Config, reg: &Region) -> Result<()> {
+    let available = shipcat_filebacked::available(conf, &reg)?;
+
+    let len = available.len();
+    let pool = ThreadPool::new(16); // 16 manifests per worker
```
Comment (gleaver) [inline shipcat_cli/src/validate.rs]: stick magic number in a named constant?

File: shipcat_cli/src/validate.rs [hunk 2]
```diff
@@ -1,5 +1,63 @@
...
+
+/// Validate all manifests in a service directory for ALL regions
+///
+/// This is meant to replace a for loop over shipcat list-regions
+pub fn all_manifests() -> Result<()> {
+    use crate::ConfigType;
+    let regions = Config::read()?.list_regions();
+
+    let len = regions.len();
+    let pool = ThreadPool::new(len); // all regions in parallel
```
Comment (gleaver) [inline shipcat_cli/src/validate.rs]: Is this additional threadpool actually doing anything useful?

Comment (clux) [reply shipcat_cli/src/validate.rs]: there's two disjoint pools atm, one pool makes 16 more.

Comment (clux) [reply shipcat_cli/src/validate.rs]: moved to rayon.

**Pull request merged**